packages/core/src/useTextField/useTextField.ts

```diff
@@ -1,191 +1,52 @@
-   */
-  placeholder?: string | undefined;
-
-  /**
-   * Autocomplete hint for the input field.
-   */
-  autocomplete?: string | undefined;
-
-  /**
-   * Whether the field is required.
-   */
-  required?: boolean;
-
-  /**
-   * Whether the field is readonly.
-   */
-  readonly?: boolean;
-
-  /**
-   * Whether the field is disabled.
-   */
-  disabled?: boolean;
-
-  /**
-   * Schema for field validation.
-   */
-  schema?: StandardSchema<string>;
-
-  /**
-   * Whether to disable HTML5 validation.
-   */
-  disableHtmlValidation?: TransparentWrapper<boolean>;
 }
 
 export function useTextField(_props: Reactivify<TextFieldProps, 'schema'>) {
   const props = normalizeProps(_props, ['schema']);
-  const inputId = useUniqId(FieldTypePrefixes.TextField);
-  const inputEl = shallowRef<HTMLInputElement>();
-  const field = useFormField<string | undefined>({
+
+  const state = useFieldState<string | undefined>({
     path: props.name,
     initialValue: toValue(props.modelValue) ?? toValue(props.value),
     disabled: props.disabled,
     schema: props.schema,
   });
 
-  const { validityDetails } = useInputValidity({ inputEl, field, disableHtmlValidation: props.disableHtmlValidation });
-  const { fieldValue, setValue, setTouched, errorMessage, isDisabled } = field;
-  const { labelProps, labelledByProps } = useLabel({
-    for: inputId,
-    label: props.label,
-    targetRef: inputEl,
-  });
-
-  const { descriptionProps, describedByProps } = createDescribedByProps({
-    inputId,
-    description: props.description,
-  });
-
-  const { accessibleErrorProps, errorMessageProps } = useErrorMessage({
-    inputId,
-    errorMessage,
-  });
-
-  const handlers: InputEvents = {
-    onInput(evt) {
-      setValue((evt.target as HTMLInputElement).value);
-    },
-    onChange(evt) {
-      setValue((evt.target as HTMLInputElement).value);
-    },
-    onBlur() {
-      setTouched(true);
+  const field = useFormField(
+    {
+      label: props.label,
+      description: props.description,
     },
-  };
+    state,
+  );
+
+  const inputId = useUniqId(FieldTypePrefixes.TextField);
 
-  const inputProps = useCaptureProps(() => {
-    return {
-      ...propsToValues(props, ['name', 'type', 'placeholder', 'autocomplete', 'required', 'readonly']),
-      ...labelledByProps.value,
-      ...describedByProps.value,
-      ...accessibleErrorProps.value,
-      ...handlers,
-      id: inputId,
-      value: fieldValue.value,
-      maxlength: toValue(props.maxLength),
-      minlength: toValue(props.minLength),
-      disabled: isDisabled.value ? true : undefined,
-      // Maybe we need to find a better way to serialize RegExp to a pattern string
-      pattern: inputEl.value?.tagName === 'TEXTAREA' ? undefined : toValue(props.pattern)?.toString(),
-    };
-  }, inputEl);
+  const { inputEl, inputProps } = useTextControl(props, { state, field, inputId });
 
   if (__DEV__) {
-    registerField(field, 'Text');
+    registerField(state, 'Text');
   }
```

Comment:
Do we want to move this to `useTextControl` as well?

That way the devtools can pick up the input when using the `injection` method